docs/qaoa/qaoa_maxcut.ipynb

```diff
@@ -0,0 +1,574 @@
+    ")\n",
+    "print(f\"Alpha = {round(alpha_value, 3)} Beta = {round(beta_value, 3)}\")\n",
+    "print(f\"Estimated cost: {estimate_cost(working_graph, sample_results)}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "rBmCr_DCsbtf"
+   },
+   "source": [
+    "## 4. Outer loop optimization\n",
+    "Now that we can compute the cost function, we want to find the optimal cost. There are lots of different techniques to choose optimal parameters for the `qaoa_circuit`. Since there are only two parameters here ($\\alpha$ and $\\beta$), we can keep things simple and sweep over incremental pairings using `np.linspace` and track the minimum value found along the way."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "ma0pVZwSThQx"
+   },
+   "outputs": [],
+   "source": [
+    "# Set the grid size = number of points in the interval [0, 2π).\n",
+    "grid_size = 5\n",
+    "\n",
+    "exp_values = np.empty((grid_size, grid_size))\n",
+    "par_values = np.empty((grid_size, grid_size, 2))\n",
+    "\n",
+    "for i, alpha_value in enumerate(np.linspace(0, 2 * np.pi, grid_size)):\n",
+    "    for j, beta_value in enumerate(np.linspace(0, 2 * np.pi, grid_size)):\n",
+    "        samples = sim.sample(\n",
+    "            qaoa_circuit,\n",
+    "            params={alpha: alpha_value, beta: beta_value},\n",
+    "            repetitions=20000,\n",
+    "        )\n",
+    "        exp_values[i][j] = estimate_cost(working_graph, samples)\n",
+    "        par_values[i][j] = alpha_value, beta_value"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "vp-LmYLnvkzM"
+   },
+   "source": [
+    "We can now visualize the cost as a function of $\\alpha$ and $\\beta$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "ZdSqBSuNuckY"
+   },
+   "outputs": [],
+   "source": [
+    "plt.title(\"Heatmap of QAOA Cost Function Value\")\n",
+    "plt.xlabel(r\"$\\alpha$\")\n",
+    "plt.ylabel(r\"$\\beta$\")\n",
+    "plt.imshow(exp_values)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "d6dd7eb92995"
+   },
+   "source": [
+    "This heatmap is coarse because we selected a small `grid_size`. To see more detail in the heatmap, one can increase the `grid_size`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "BzwnTYWpuKZM"
+   },
+   "source": [
+    "## 5. Compare cuts\n",
+    "\n",
+    "We now compare the optimal cut found by QAOA to a randomly selected cut. The helper function draws the `working_graph` and colors nodes in different sets different colors. Additionally, we print out the cost function for the given cut."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "6nD1YQr39KOI"
+   },
+   "outputs": [],
+   "source": [
+    "def output_cut(S_partition):\n",
```

Comment:
nit: Correct capitalization for style would be all lowercase
Also could use type info.